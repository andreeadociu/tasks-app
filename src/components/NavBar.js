import React from "react";
import { Link } from "react-router-dom";

function NavBar(props) {
  return (
    <nav data-testid="navbar">
      <div className="nav-wrapper">
        <Link to="/" className="brand-logo">
          {props.title}
        </Link>
      </div>
    </nav>
  );
}

export default NavBar;
