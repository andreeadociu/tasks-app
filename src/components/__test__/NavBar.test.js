import React from "react";
import ReactDOM from "react-dom";
import NavBar from "../NavBar";
import { BrowserRouter as Router } from "react-router-dom";

import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

it("renders without crashing", () => {
  const nav = document.createElement("nav");
  ReactDOM.render(
    <Router>
      <NavBar />
    </Router>,
    nav
  );
});

it("renders navbar correctly", () => {
  const { getByTestId } = render(
    <Router>
      <NavBar title="Test" />
    </Router>
  );

  expect(getByTestId("navbar").textContent).toBe("Test");
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <Router>
        <NavBar title="Test" />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
