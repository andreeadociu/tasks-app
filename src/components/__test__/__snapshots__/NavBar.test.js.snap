// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  data-testid="navbar"
>
  <div
    className="nav-wrapper"
  >
    <a
      className="brand-logo"
      href="/"
      onClick={[Function]}
    >
      Test
    </a>
  </div>
</nav>
`;
